quattor/blockdevices.pan

```diff
@@ -103,10 +103,30 @@ type blockdevices_file_type = {
 type raid_device_path = string with exists ("/system/blockdevices/hwraid/"
     + SELF) || is_card_port (SELF);
 
+@documentation{
+    blockdevices correct device check based on device size
+}
+type blockdevices_correct_size = {
+    @{allowed absolute margin (in MiB) compared to configured size}
+    "diff" ? long(0..)
+    @{allowed relative margin compared to configured size}
+    "fraction" ? double
+};
+
+@documentation{
+    blockdevices correct device check
+}
+type blockdevices_correct = {
+    @{size-based correctness check}
+    "size" ? blockdevices_correct_size
+};
+
 type blockdevices_disk_type = {
     "device_path" ? raid_device_path
     "label" : string with match (SELF, ("^(none|msdos|gpt|aix|bsd)$"))
     "readahead" ? long # Blocks for readahead
+    @{enable correctness checks}
+    "correct" ? blockdevices_correct
```

Comment:
I think the annotation is fine, it's the field name that seemed suboptimal because correct(ness) is an adjective or verb, while a field represents an object (noun).  My ideal suggestion is ```attributes_to_validate``` but that might be a bit long so happy for shorter alternatives.